### PR TITLE
Fix #5326, use old style API to behave consistently in different runtime

### LINKED ISSUE #5326: Error in webpack-sources
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
TypeError: item.source is not a function
```
TypeError: item.source is not a function
    at /Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:23:50
    at Array.map (native)
    at ConcatSource.source (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:22:24)
    at /Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:23:50
    at Array.map (native)
    at ConcatSource.source (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:22:24)
    at /Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:23:50
    at Array.map (native)
    at ConcatSource.source (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:22:24)
    at CachedSource.source (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/webpack-sources/lib/CachedSource.js:28:44)
    at Compiler.writeOut (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/lib/Compiler.js:346:27)
    at require.forEach (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/lib/Compiler.js:336:21)
    at /Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/async/dist/async.js:3083:16
    at eachOfArrayLike (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/async/dist/async.js:1003:9)
    at eachOf (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/async/dist/async.js:1051:5)
    at Object.eachLimit (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/node_modules/async/dist/async.js:3145:5)
    at Compiler.emitFiles (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack/lib/Compiler.js:325:21)
    at Immediate.<anonymous> (/Users/u_021ly/Desktop/projects/arui-feather/node_modules/webpack-dev-middleware/node_modules/memory-fs/lib/MemoryFileSystem.js:288:4
```
This is Travis build https://travis-ci.org/alfa-laboratory/arui-feather/jobs/255237762#L718
**If the current behavior is a bug, please provide the steps to reproduce.**
```
git clone https://github.com/alfa-laboratory/arui-feather.git
git checkout feat/add-md-lint
npm i
npm run gemini-ci
```

**What is the expected behavior?**
Runs without errors.
**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
npm -v 5.3.0
node -v 6.10.1
webpack 3.3.0

### PR DESCRIPTION
**Pull request by benjycui**
**What kind of change does this PR introduce?**

Fix #5326

**Did you add tests for your changes?**

Yes.

**If relevant, link to documentation update:**

**Summary**

Just a bugfix.

**Does this PR introduce a breaking change?**

Nope.

**Other information**

### CODE REVIEW

File: lib/Stats.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const identifierUtils = require("./util/identifier");
 const optionsOrFallback = function() {
 	let optionValues = [];
 	optionValues.push.apply(optionValues, arguments);
-	return optionValues.find(optionValue => typeof optionValue !== "undefined");
+	return optionValues.filter(optionValue => typeof optionValue !== "undefined")[0];
```
Comment (sokra) [inline lib/Stats.js]: This neither should make a difference.

File: lib/Chunk.js [hunk 2]
```diff
@@ -160,7 +160,7 @@ class Chunk {
 	}
 
 	mapModules(fn) {
-		return Array.from(this._modules, fn);
+		return Array.from(this._modules).map(fn);
```
Comment (sokra) [inline lib/Chunk.js]: This should not make any difference.


from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

> More clearly, Array.from(obj, mapFn, thisArg) has the same result as Array.from(obj).map(mapFn, thisArg), except that it does not create an intermediate array.

**Comment by benjycui** [comment]
Yep, it is weirdo, and I cannot provide a minimal re-producible repo, so just let @cbrwizard @unfalse @Smolations @stepancar @Donmclean who met this issue to try my patch first.

**Comment by benjycui** [comment]
I dive deeper, and found that my code require a polyfill which had changed `Array.prototype.find` and `Array.from`. https://github.com/montagejs/collections/blob/master/shim-array.js

I didn't require this polyfill directly, I just require gh-pages, and then `gh-pages > q-io > collections`.

I just update gh-pages to latest version which dropped q-io, and then everything works fine.

I think that @cbrwizard @unfalse @Smolations @stepancar @Donmclean may require some polyfill too, even thought they don't know it. So the behavior of API had been changed, and then break webpack's code.

@sokra of course, you can close this PR, for this is not the bug of webpack. If you merge this PR, will help those people who even don't know they had import a polyfill. Anyway, this PR will not hurt, as you said: This should not make any difference.

**Comment by sokra** [comment]
I close it. `Array.from(a, b)` is faster than `Array.from(a).map(b)`

**Comment by sokra** [comment]
btw. it's actually super bad from montagejs/collections to add broken polyfills. You should file an issue there.

**Pull request closed without merging**